### DISCUSSION TITLE
-D_GNU_SOURCE break the build on Linux?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#140] [General]
Cygwin needs `-D_GNU_SOURCE` otherwise it will fail to compile. Currently I reused the Makefiles of Linux for Cygwin. After the commit https://github.com/objeck/objeck-lang/commit/75bd19035bf9af77ecb4352155475a4c9cffcdea I think Cygwin will need it own Makefiles, the build script `deploy_cygwin.sh` also has to be modified to use the new Cygwin specific Makefiles.

**Comment by objeck** [comment]
Ok, let's add new Cygwin build files instead. This was done for Clang64, UCRT64, and Linux ARM64.

**Reply by ghost** [reply]
You will do it or I do a PR? It's simply copy paste so it's very quick.

**Reply by objeck** [reply]
A PR would be great, as I do not have Cygwin 64 installed. I will do that over the weekend.
